.github/workflows/precompiled.yaml

```diff
@@ -120,42 +152,28 @@ jobs:
         id: set_kernel_version
         env:
           BASE_TARGET: "jammy"
-          DIST: "ubuntu22.04"
+          DIST: ${{ matrix.dist }}
         run: |
           echo "matrix_values_not_empty=0" >> $GITHUB_OUTPUT
 
           kernel_flavors_json='${{ needs.set-driver-version-matrix.outputs.kernel_flavors }}'
-          kernel_flavors=$(echo "$kernel_flavors_json" | jq -r '.[]')
+          KERNEL_FLAVORS=($(echo "$kernel_flavors_json" | jq -r '.[]'))
           driver_branch_json='${{ needs.set-driver-version-matrix.outputs.driver_branch }}'
-          driver_branch=$(echo "$driver_branch_json" | jq -r '.[]')
-
-          kernel_versions=()
-          for kernel_flavor in $kernel_flavors; do
-            # FIXME -- remove if condition, once azure kernel upgrade starts working
-            if [[ "$kernel_flavor" == "azure" ]]; then
-              echo "skipping azure kernel testing"
-              continue
-            fi
-            for DRIVER_BRANCH in $driver_branch; do
-              source ./tests/scripts/findkernelversion.sh "$BASE_TARGET" "${kernel_flavor}" "$DRIVER_BRANCH" "$DIST"
-              if [[ "$should_continue" == true ]]; then
-                echo "matrix_values_not_empty=1" >> $GITHUB_OUTPUT
-                break
-              fi
-            done
-            if [[ "$should_continue" == false ]]; then
-                echo "Skipping e2e tests for the following driver tag: ${KERNEL_VERSION}-${kernel_flavor}-${DIST}"
-            else
-              KERNEL_VERSION=$(echo "$KERNEL_VERSION" | tr -d ' \n')
-              kernel_versions+=("$KERNEL_VERSION")
-               echo "Adding the following tag to the e2e test matrix: ${KERNEL_VERSION}-${kernel_flavor}-${DIST}"
-            fi
-          done
+          DRIVER_BRANCHES=($(echo "$driver_branch_json" | jq -r '.[]'))
 
+          source ./tests/scripts/ci-precompiled-helpers.sh
+          KERNEL_VERSIONS=($(get_kernel_versions_to_test $BASE_TARGET KERNEL_FLAVORS[@] DRIVER_BRANCHES[@] $DIST))
+          if [ -z "$KERNEL_VERSIONS" ]; then
+            # no new kernel release
+            echo "Skipping e2e tests"
+            exit 0
+          fi
+          echo "matrix_values_not_empty=1" >> $GITHUB_OUTPUT
```

Comment:
This command is a duplicate of line 176. We can remove this line.

Reply:
removed . Thanks